src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -321,6 +323,101 @@ network::network(program::ptr program, uint16_t stream_id)
 network::network(program::ptr program, stream::ptr stream, uint16_t stream_id)
     : network(program, stream, false, stream_id == 0) {}
 
+network::network(cldnn::BinaryInputBuffer& ib, stream::ptr stream, engine& engine, uint16_t stream_id)
+    : _program(nullptr)
+    , _engine(ib.get_engine())
+    , _stream(stream)
+    , _memory_pool(new memory_pool(engine))
+    , _internal(false)
+    , _is_primary_stream(false)
+    , _reset_arguments(true) {
+    net_id += 1;
```

Comment:
I believe net_id is always 1 here which is unexpected.

Reply:
https://github.com/openvinotoolkit/openvino/pull/13986/files#r1025253714
I added a new function get_new_net_id() to emit an unique id, and applied it to network ctors. Thank you.